crypto/x509/x_all.c

```diff
@@ -435,23 +435,25 @@ ASN1_OCTET_STRING *X509_digest_sig(const X509 *cert)
 int X509_CRL_digest(const X509_CRL *data, const EVP_MD *type,
                     unsigned char *md, unsigned int *len)
 {
-    if (type == EVP_sha1() && (data->flags & EXFLAG_SET) != 0
+    if (type != NULL
+            && EVP_MD_is_a(SN_sha1)
```

Comment:
I suggest you try to compile this :)

Reply:
Yep.. This was a WIP branch that I half did.. Fixed it up now.. It was more so that @t8m didnt have to redo some of the code.